File: Data Structures & Algorithms/longest-palindromic-substring/submission-3.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        res = ""

        def expand(i, j):
            while i >= 0 and j < n  and s[i] == s[j]:
                i -= 1
                j += 1
            return s[i+1 : j]

        for i in range(n):
            p1 = expand(i, i)
            p2 = expand(i, i+1)
            
            if len(p1) > len(res):
                res = p1

            if len(p2) > len(res):
                res = p2

        return res

File: Data Structures & Algorithms/n-queens/submission-3.go
func solveNQueens(n int) [][]string {
	col := make([]bool, n)
    posDiag := make([]bool, 2*n)
    negDiag := make([]bool, 2*n)

    var res [][]string
    board := make([][]rune, n)

    for i := range board {
        board[i] = make([]rune, n)
        for j := range board[i] {
            board[i][j] = '.'
        }
    }
	
	var backtrack func(r int)
 	backtrack = func(r int){
		if r == n {
            solution := make([]string, n)
            for i := range board {
                solution[i] = string(board[i])
            }
            res = append(res, solution)
            return
        }

		for c := 0; c < n; c++ {
            if col[c] || posDiag[r+c] || negDiag[r-c+n] {
                continue
            }
			
			col[c] = true
            posDiag[r+c] = true
            negDiag[r-c+n] = true
            board[r][c] = 'Q'
			
			backtrack(r + 1)

			col[c] = false
            posDiag[r+c] = false
            negDiag[r-c+n] = false
			board[r][c] = '.'
		}
	}

	backtrack(0)
	return res
}
